playkit/src/main/java/com/kaltura/playkit/player/PKSubtitlePosition.java

```diff
@@ -0,0 +1,182 @@
+        }
+
+        setHorizontalPositionLevel(horizontalPosition, horizontalAlignment);
+        setVerticalPositionLevel(verticalPosition);
+        lineType = Cue.LINE_TYPE_FRACTION;
+        return this;
+    }
+
+    /**
+     * Set the subtitle position only in Vertical direction (Up or Down) on the video frame. This method only allows to move in Y - coordinate
+     * To set the subtitle any where on the video frame use {@link PKSubtitlePosition#setPosition(int, int, Layout.Alignment)}
+     *
+     * @param verticalPositionPercentage Top(10 is for 10%) to Bottom(100 is for 100%)
+     * @return PKSubtitlePosition
+     */
+    public PKSubtitlePosition setVerticalPosition(int verticalPositionPercentage) {
+        float verticalPosition = checkPositionPercentageLimit(verticalPositionPercentage);
+
+        setVerticalPositionLevel(verticalPosition);
+        subtitleHorizontalPosition = Layout.Alignment.ALIGN_CENTER;
+        horizontalPositionPercentage = Cue.DIMEN_UNSET;
+        lineType = Cue.LINE_TYPE_FRACTION;
+        return this;
+    }
+
+    /**
+     * If `overrideInlineCueConfig` is false that mean; app does not want to override the inline Cue configuration.
+     * App wants to go with Cue configuration.
+     * BUT Beware that it will call {@link PKSubtitlePosition#setOverrideInlineCueConfig(boolean)} with false value
+     * means after that in next call, app needs to {@link PKSubtitlePosition#setOverrideInlineCueConfig(boolean)}
+     * with another value.
+     *
+     * OTHERWISE
+     *
+     * If `overrideInlineCueConfig` is true then it will move subtitle to Bottom-Center which is a standard position for it
+     *
+     * @return PKSubtitlePosition
+     */
+    public PKSubtitlePosition setToDefaultPosition(boolean overrideInlineCueConfig) {
+
+        if (!overrideInlineCueConfig) {
+            setOverrideInlineCueConfig(false);
+            return this;
+        }
+
+        subtitleHorizontalPosition = null;
+        verticalPositionPercentage = Cue.DIMEN_UNSET;
+        horizontalPositionPercentage = Cue.DIMEN_UNSET;
+        lineType = Cue.TYPE_UNSET;
+        return this;
+    }
+
+    /**
+     * If `overrideInlineCueConfig` is set to true, player will ignore the existing values coming in Cue Settings
+     * and will always use the given Cue Settings
+     *
+     * @param overrideInlineCueConfig true or false
+     * @return PKSubtitlePosition
+     */
+    public PKSubtitlePosition setOverrideInlineCueConfig(boolean overrideInlineCueConfig) {
+        this.overrideInlineCueConfig = overrideInlineCueConfig;
+        return this;
+    }
+
+    private float checkPositionPercentageLimit(int positionPercentage) {
+        float position;
+        if (positionPercentage < positionLowerLimit || positionPercentage > Consts.DEFAULT_MAX_SUBTITLE_POSITION) {
+            position = Cue.DIMEN_UNSET;
+        } else {
+            position = (float) positionPercentage / Consts.PERCENT_FACTOR_FLOAT;
```

Comment:
The `(float)` cast is not needed, since `PERCENT_FACTOR_FLOAT` is already float.